### ISSUE TITLE
Can't use search in channel !

### ISSUE THREAD

**Issue by AizeLeOuf** [gnello/php-mattermost-driver#27]
According to https://api.mattermost.com/#tag/channels%2Fpaths%2F~1teams~1%7Bteam_id%7D~1channels~1search%2Fpost

a HTTP Post was expected but in the code there is a get :

https://github.com/gnello/php-mattermost-driver/blob/56bea0dc0e2a6cd92117ceb5842ea8b07c5691ec/src/Models/ChannelModel.php#L322

About that, can't find an example, to inject data in post body.. :'( can you provide an example for that ?

Best regards,

And many thanks to share your work !

**Comment by gnello** [comment]
Hi!
Thanks for your contribution on this error!

This issue was solved (by you :smile: ) in #28.

I'm going to provide an example in the readme, thanks for pointing that out.

**Comment by gnello** [comment]
I've just added this example in the readme:

```php
//Search a channel
$result = $driver->getChannelModel()->searchChannels($teamId, [
    'term' => "full or partial name or display name of channels"
]);
```
Let me know if it helps.

**Comment by AizeLeOuf** [comment]
Just perfect !! Many thanks :)

Another last question, have you plan to support methods added in >5, for example https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1%7Bchannel_id%7D~1privacy%2Fput ?

**Comment by gnello** [comment]
Sure, I'm going to add support for all new method as soon as I can, I hope I can do it even this weekend.

I'll keep you updated.

**Comment by gnello** [comment]
Hi,
I've just released the 2.10.0 version that it should be up to date with the current version of mattermost server.
Let me know if you notice any inaccuracies.

Thanks.

**Comment by gnello** [comment]
Hi, 
I'm closing this issue assuming everything's ok 🙂. 
Please, open a new issue if you have any further requests.

Thank you!
